lib/connectors_async/job.rb

```diff
@@ -86,5 +86,9 @@ def is_finished?
 
       [ConnectorsShared::JobStatus::FINISHED, ConnectorsShared::JobStatus::FAILED].include?(status)
     end
+
+    def should_throttle?
```

Comment:
I think my only recommended change would be to change this name. It implies slowing activity, but we want to completely stop activity until later.

How about:
```suggestion
    def should_wait?
```

Reply:
Good call! Renamed to `should_wait?`